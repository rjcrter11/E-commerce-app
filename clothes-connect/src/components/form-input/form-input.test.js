import React from 'react';

import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import FormInput from './form-input.component';

describe('FormInput component', () => {
    let wrapper;
    let enzymeWrapper;
    let mockHandleChange;

    beforeEach(() => {
        mockHandleChange = jest.fn()

        const mockProps = {
            label: 'email',
            value: 'test@gmail.com',
            handleChange: mockHandleChange
        };

        wrapper = create(<FormInput {...mockProps} />)
        enzymeWrapper = shallow(<FormInput  {...mockProps} />)
    });

    it('should render FormInput component', () => {
        expect(wrapper.toJSON()).toMatchSnapshot()
    });
    it('should call handleChange method when input changes', () => {
        enzymeWrapper.find('[data-testid="form-input"]').simulate('change');
        expect(mockHandleChange).toHaveBeenCalled();
    })
    it('should render form input label if there is a label', () => {
        expect(enzymeWrapper.exists('[data-testid="input-label"]')).toBe(true)
    });

    it('should not render form input label if there is no label', () => {
        const mockNewProps = {
            label: '',
            value: 'test@gmail.com',
            handleChange: mockHandleChange
        };
        const newWrapper = shallow(<FormInput {...mockNewProps} />);
        expect(newWrapper.exists('[data-testid="input-label"]')).toBe(false)
    })
})