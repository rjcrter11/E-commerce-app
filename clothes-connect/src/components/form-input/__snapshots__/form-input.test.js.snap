// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render FormInput component 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    data-testid="form-input"
    onChange={[MockFunction]}
    value="test@gmail.com"
  />
  <label
    className="shrink form-input-label"
    data-testid="input-label"
  >
    email
  </label>
</div>
`;
